response-to-audit-report.md

```diff
@@ -0,0 +1,2410 @@
+> 
+> The argument by which the `Contract` in the `txInfoData` list corresponds to the given hash needs to be made explicit.
+
+Commit [b6366efc](https://github.com/input-output-hk/marlowe-cardano/commit/b6366efc9f8df38bfe9232f81e6caf2c7f11ffa8) makes this hash correspondence explicit in the Marlowe Cardano specification and adds a code snippet concretizing it, namely that the hash and continuation must be present in the script context datum map.
+
+
+### 2.5.4 Merkleization section not detailed enough *(Severity: Low)*
+
+> **File `marlowe-cardano-specification.md`, Section `Merkleization`**
+> 
+> This section is too terse. It needs to explain what Merkleization is, and to motivate why it is needed.
+> 
+> When explaining how it works, it needs to make explicit that only the `Case` type is modified, and that in the semantics, only the `Input` type is modified. It needs to explain why the `Input` type needs to carry a hash and a contract, and why the evaluation of the contract is changed as described.
+
+Commit [a9d40432](https://github.com/input-output-hk/marlowe-cardano/commit/a9d4043230cb4f240fab2d816bed6a3cab4f5053) adds to the Marlowe Cardano specification a paragraph elaborating on mechanics and motivation for Merkleization, namely the ability to handle Marlowe contract that are too large to store in a datum.
+
+
+### 2.5.5 Unnecessary constraint *(Severity: Low)*
+
+> **File `marlowe-cardano-specification.md`, `Constraint 12. Merkleized continuations`** 
+> 
+> This constraint is unnecessary to have in the Marlowe validator, since the construction of the arguments for evaluation of the Marlowe contract would fail. However, it would be useful to have it appear in the specification for users to be aware of it when crafting transactions. A note to motivate the presence of the constraint could be helpful.
+
+Commit [fda1d3e8](https://github.com/input-output-hk/marlowe-cardano/commit/fda1d3e862fc8f8c31ff479fbd338226e661eb0b) to the Marlowe Cardano specification clarifies that the constraint is supplementary (informational) in nature.
+
+
+### 2.5.6 Asymmetry between role and wallet payouts *(Severity: Low)*
+
+> **File `marlowe-cardano-specification.md`, `Constraint 15`, **
+> 
+> The marlowe validator allows multiple outputs to be paid to a wallet, but it demands that a single output exists when paying to a role instead. The motivation to use different approaches needs to be documented. This is implemented in `Scripts.hs` at line 371, in function `payoutConstraints`.
+
+The asymmetric is a convenience and practicality related to the manner in which coin selection and transaction balancing typically occurs nowadays in wallets. Commit [4e81470d](https://github.com/input-output-hk/marlowe-cardano/commit/4e81470de90ddd6858c3ad8aace96e90df5a86ae) adds to the Marlowe Cardano specification a paragraph justifying this asymmetry in payout style and a comment to the Marlowe semantics validator along the same lines.
```

Comment:
> Also, is this because role is done through payout validator?

Mostly, but this is subjective. Payouts to addresses and payouts to the role validator are definitely handled differently, partly for historical reasons and partly conceptually. We may want to change this in a future version of the validator.

> Does the payout validator allow several outputs in the redeem step?

Yes, the payout validator can redeem several UTxOs in the same transaction, subject to the Plutus execution budget.